src/operator/timeout.ts

```diff
@@ -20,7 +20,8 @@ export function timeout<T>(this: Observable<T>, due: number | Date,
                            scheduler: Scheduler = async): Observable<T> {
   const absoluteTimeout = isDate(due);
   const waitFor = absoluteTimeout ? (+due - scheduler.now()) : Math.abs(<number>due);
-  const error = errorToSend || new TimeoutError();
+  const error = errorToSend instanceof Error ? errorToSend : new TimeoutError(errorToSend);
```

Comment:
What about instead checking if it's a string? if yes, wrap, if no, use as-is. Otherwise if someone provides their own object that doesn't extend `Error`, it would log `TimeoutError: [object Object]`